doc/stp/MSTP_HLD.md

```diff
@@ -0,0 +1,822 @@
+
+# Table of Contents
+
+* [Scope](#scope)
+* [Background](#background)
+* [Abbreviations](#abbreviations)
+* [Overview](#overview)
+* [Introduction](#introduction)
+* [Requirements](#requirements)
+* [Architecture Design](#architecture-design)
+  - [STP Container](#stp-container)
+  - [SWSS Container](#swss-container)
+  - [CoPP Configurations](#copp-configurations)
+* [Database Changes](#database-changes)
+  - [CONFIG DB](#config-db)
+  - [APP DB](#app-db)
+* [SAI](#sai)
+* [Additional Features](#additional-features)
+* [Sequence Diagrams](#sequence-diagrams)
+  - [MSTP global enable](#mstp-global-enable)
+  - [MSTP global disable](#mstp-global-disable)
+  - [MSTP region name/version change](#mstp-region-nameversion-change)
+  - [Instance creation](#instance-creation)
+  - [Instance deletion](#instance-deletion)
+  - [Add VLAN to instance](#add-vlan-to-instance)
+  - [Del VLAN from instance](#del-vlan-from-instance)
+* [Configuration Commands](#configuration-commands)
+  - [Global Level](#global-level)
+  - [Region Level](#region-level)
+  - [Instance, Interface Level](#instance-interface-level)
+  - [Show Commands](#show-commands)
+  - [Clear Commands](#clear-commands)
+  - [Debug Commands](#debug-commands)
+  - [Disabled Commands](#disabled-commands)
+* [YANG Model](#yang-model)
+* [Rest API Support](#rest-api-support)
+* [Warm Boot](#warm-boot)
+* [Testing Requirements](#testing-requirements)
+  - [Unit test cases](#unit-test-cases)
+* [References](#references)
+
+# Scope
+This document describes the High Level Design of Multiple Spanning Tree Protocol.
+
+# Background
+This HLD is based on the design provided by PVST HLD.
+
+PVST HLD: https://github.com/sonic-net/SONiC/pull/386
+
+# Abbreviations
+
+|**Term**|**Meaning**|
+| :- | :- |
+|MSTP|Multiple Spanning Tree Protocol|
+|PVST|Per VLAN Spanning Tree|
+|STP|Spanning Tree Protocol|
+|VLAN|Virtual Local Area Network|
+|MSTI|Multiple Spanning Tree Instance|
+|CIST|Common Internal Spanning Tree|
+|CST|Common Spanning Tree|
+|IST|Internal Spanning Tree|
+|BPDU|Bridge Protocol Data Unit|
+|RSTP|Rapid Spanning Tree Protocol|
+|VID|VLAN identifier|
+|MSTID|Multiple Spanning Tree Identifier|
+|CoPP|Control Plane Policing|
+
+# Overview
+Multiple Spanning Tree Protocol (MSTP) enhances the Spanning Tree Protocol (STP) by enabling the creation of multiple spanning tree instances within a network. It provides a mechanism to map VLANs to specific spanning tree instances which offers network segmentation and improved control over traffic flow.
+
+# Introduction
+Spanning Tree Protocol (STP) prevents bridge looping on LANs that include redundant links. Per VLAN Spanning Tree (PVST) is a modification of STP that allows for running separate instances of spanning tree for each VLAN. 
```

Comment:
Also which STP implementation, you are planning to use?  i.e.) sonic stp repro https://github.com/sonic-net/sonic-stp introduced by Broadcom  is kind of orphaned not being used and there is Linux natively MSTP support or planning custom MSTP implementation from your side?

Reply:
Currently, We are planning to use existing STP repo.  But if you have any suggestions or specific use case in this regard we can have discussion on it as well.